ui/src/cohort.ts

```diff
@@ -213,6 +217,64 @@ export function upgradeCriteria(
   }
 }
 
+export interface PredefinedCriteria {
+  id: string;
+  occurrence: string;
+  filter?: Filter;
+}
+
+export type OccurrenceFilters = {
+  id: string;
+  name: string;
+  attributes: string[];
+  filters: Filter[];
+};
+
+export function getOccurrenceList(
+  source: Source,
+  selectedCriteria: Set<string>,
+  userCriteria?: tanagraUI.UICriteria[],
+  predefinedCriteria?: PredefinedCriteria[]
+): OccurrenceFilters[] {
+  const occurrences = new Map<string, Filter[]>();
+  const addFilter = (occurrence: string, filter?: Filter | null) => {
+    if (!occurrences.has(occurrence)) {
+      occurrences.set(occurrence, []);
+    }
+    if (filter) {
+      occurrences.get(occurrence)?.push(filter);
+    }
+  };
+
+  predefinedCriteria
```

Comment:
Does predefined here refer to prepackaged data features? Or what's the difference between predefinedCriteria and userCriteria?

Reply:
Yes, prepackaged. I'm just moving towards clearer naming: predefined vs. user defined.